scripts/babel-relay-plugin/src/GraphQLPrinter.js

```diff
@@ -290,6 +290,14 @@ function printFieldsAndFragments(
         fragments.push(printQueryFragment(selection, options));
       } else if (selection.kind === kinds.FIELD) {
         fields.push(selection);
+
+        if (
+          isConnectionType(type) &&
```

Comment:
this should probably be `!!getConnectionMetadata(...)` in order to avoid adding `pageInfo` on fields that look like connections but don't follow the Relay connection spec.